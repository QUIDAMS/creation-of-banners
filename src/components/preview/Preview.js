import React from 'react';
import PropTypes from 'prop-types';

function backgroundValue(banner) {
  if(banner.image) {
    return(`center/contain url("${banner.image}") no-repeat`);
  } else if (banner.color){
    return(banner.color);
  } else {
    return(`linear-gradient(${banner.topColorGradient}, ${banner.bottomColorGradient})`);
  }
}

const Preview = React.forwardRef(({banner}, ref) => {
  const divStyle = {
    background: backgroundValue(banner),
    position: 'relative',
    width: 250,
    height: 350,
    display: 'inline-block',
    border: '1px solid black',
    fontSize: 35,
    fontWeight: 'bold',
  }

  const textStyle = {
    position: 'absolute',
    left: 20,
    bottom: 5,
    margin: 0,
    color: 'black'
  }

  const paragraphStyle = {
    margin: 2,
  }

  return(
    <div className='preview'>
      <a
        href={banner.url}
        target='_blank'
        rel='noreferrer'
      >
        <div style={divStyle} ref={ref}>
          <div style={textStyle}>
            <p style={paragraphStyle} className='preview__text-line'>{banner.line1}</p>
            <p style={paragraphStyle} className='preview__text-line'>{banner.line2}</p>
            <p style={paragraphStyle} className='preview__text-line'>{banner.line3}</p>
          </div>
        </div>
      </a>
    </div>
  );
});

Preview.propTypes = {
  banner: PropTypes.object,
}

export default Preview;
