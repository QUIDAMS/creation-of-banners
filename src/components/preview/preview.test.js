import React from 'react';
import Preview from './Preview';
import {shallow} from 'enzyme';

describe('Testing Preview', () => {
  const props = {
    banner: {
      image: '',
      line1: '',
      line2: '',
      line3: '',
      color: '',
      topColorGradient: '',
      bottomColorGradient: '',
      url: '',
    }
  }
  it('render correctly correctly', () => {
    const editor = shallow(<Preview banner={props}/>);
    expect(editor).toMatchSnapshot();
  });
});
