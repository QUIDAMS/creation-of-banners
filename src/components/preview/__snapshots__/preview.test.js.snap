// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Preview render correctly correctly 1`] = `
<div
  className="preview"
>
  <a
    rel="noreferrer"
    target="_blank"
  >
    <div
      style={
        Object {
          "background": "linear-gradient(undefined, undefined)",
          "border": "1px solid black",
          "display": "inline-block",
          "fontSize": 35,
          "fontWeight": "bold",
          "height": 350,
          "position": "relative",
          "width": 250,
        }
      }
    >
      <div
        style={
          Object {
            "bottom": 5,
            "color": "black",
            "left": 20,
            "margin": 0,
            "position": "absolute",
          }
        }
      >
        <p
          className="preview__text-line"
          style={
            Object {
              "margin": 2,
            }
          }
        />
        <p
          className="preview__text-line"
          style={
            Object {
              "margin": 2,
            }
          }
        />
        <p
          className="preview__text-line"
          style={
            Object {
              "margin": 2,
            }
          }
        />
      </div>
    </div>
  </a>
</div>
`;
